pages/api/schedule/ics.tsx

```diff
@@ -0,0 +1,83 @@
+import { firestore } from 'firebase-admin';
+import { NextApiRequest, NextApiResponse } from 'next';
+import initializeApi from '../../../lib/admin/init';
+import { userIsAuthorized } from '../../../lib/authorization/check-authorization';
+import { createEvent } from 'ics';
+initializeApi();
+const db = firestore();
+
+const SCHEDULE_EVENTS = '/schedule-events';
+
+async function getIcalEvent(req: NextApiRequest, res: NextApiResponse) {
+  let event = await db.collection(SCHEDULE_EVENTS).get();
+  console.log(event);
+  if (event.empty) {
+    console.log(event.size);
+    console.log(req.query.Event);
+  }
+  let eventName = '';
+  let icalEvents = [];
+  event.forEach((e) => {
```

Comment:
Nit: Avoid using the for each method and then pushing to a vector. Using [event.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/map) is more idiomatic.

Reply:
I actually tried this initially but event is a QuerySnapshot (https://firebase.google.com/docs/reference/js/firestore_.querysnapshot) and that type does not have a map function. :(

Reply:
![image](https://github.com/albertaspsc/hack-portal/assets/112734799/fd3661d0-7381-47e1-ba73-93759a215048)